pkg/action/validate.go

```diff
@@ -55,7 +55,8 @@ func requireAdoption(resources kube.ResourceList) (kube.ResourceList, error) {
 			return fmt.Errorf("could not get information about the resource %s: %w", resourceString(info), err)
 		}
 
-		requireUpdate.Append(info)
+		infoCopy := *info
```

Comment:
Does this need to be a deep copy?

Reply:
(I should mention, I want to keep this PR minimal as it needs to be a Helm 4.0.1 bugfix. Additional hardening would/should be a new PR)